Dart Challange/game4/lib/game4.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:io';
+
+void startGame() {
+  bool loopGame = true;
+  while (loopGame) {
+    stdout.writeln('Welcome to Magical Hat Game.');
+    stdout.writeln('Type \'exit\' to stop the game');
+    final resultText = 'Lance Lancelot'.toUpperCase();
+    List<String> listResultText = resultText.split('');
+    List<String> listHints = createHintText(listResultText);
+
+    loopGame = playGame(
+      resultText: listResultText,
+      hintText: listHints
+    );
+
+    stdout.writeln(loopGame ? "Play again!\n" : "\n");
+  }
+}
+
+List<String> createHintText(List<String> resultText) {
+  List<String> listHintText = [];
+  for (String character in resultText) {
+    if (character != ' ') {
+      listHintText.add('_');
+    } else {
+      listHintText.add(' ');
+    }
+  }
+  return listHintText;
+}
+
+String getUserInput() {
+  String? userInput;
+  do {
+    stdout.write("Please guess a letter: ");
+    userInput = stdin.readLineSync();
+  } while (userInput!.length != 1 && userInput != 'exit');
+  return userInput;
+}
+
+void renderHintText(List<String> hints){
+  for (String character in hints){
+    stdout.write(character);
+  }
+  print('');
```

Comment:
What does code mean? If it's useless, please remove it

Reply:
I use print('') to break a new line. But I can replace it with "\n" in code